lib/logstash/util/compress.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/environment"
+
+module LogStash::Util
+  module Zip
+
+    extend self
+
+    def extract(source, target)
+      require "zip"
+      require "fileutils"
+
+      puts "Extracting #{source} to #{target}"
+      ::Zip::File.open(source) do |zip_file|
+        zip_file.each do |file|
+          path = File.join(target, file.name)
+          FileUtils.mkdir_p(File.dirname(path))
+          zip_file.extract(file, path)
+        end
+      end
+      puts "finished"
+    end
+
+    def compress(dir, target)
+      require "zip"
+      puts "Compressing #{dir} into #{target}"
+      ::Zip::File.open(target, ::Zip::File::CREATE) do |zipfile|
+        Dir.glob("#{dir}/**/*").each do |file|
+          path_in_zip = file.gsub("#{dir}/","")
+          zipfile.add(path_in_zip, file)
+        end
+      end
+      puts "finished"
+    end
+
+  end
+
+  module Tar
+
+    extend self
+
+    def extract(file, target)
+      raise Exception.new("Not using JRuby") unless LogStash::Environment.jruby?
+      puts "Extracting #{file} to #{target}"
+      require "rubygems/package"
+      require "fileutils"
+      require "tempfile"
+      require "zlib"
+
+      Zlib::GzipReader.open(file) do |gzip_file|
+        Gem::Package::TarReader.new(gzip_file) do |tar_file|
+           tar_file.each do |entry|
+             target_path = File.join(target, entry.full_name)
+
+            if entry.directory?
+              FileUtils.mkdir_p(target_path)
+            else # is a file to be extracted
+              File.open(target_path, "wb") do |f|
+                f.write(tar_file.read)
+              end
+            end
+           end
+         end
+      end
+      puts "finished"
+    end
+
+    def compress(dir, target)
+      raise Exception.new("Not using JRuby") unless LogStash::Environment.jruby?
+      puts "Compressing #{dir} into #{target}"
+
+      require "rubygems/package"
+      require "fileutils"
+      require "tempfile"
+      require "zlib"
+
+      tar_file = Tempfile.new(File.basename(target))
```

Comment:
the `target` passed to the `compress` method will not be the actual target used, this seems confusing and it's not documented.